app/src/main/java/MVC/GlobalControllerFactory.java

```diff
@@ -58,7 +58,7 @@ public BaseController   getControllerForLifecycleOwner(UUID ownerId) {
         } else {
             Constructor<?> ctor = null;
             try {
-                ctor = controllerClass.getConstructor(owner.getClass());
+                ctor = controllerClass.getConstructor(owner.getClass().getInterfaces()[0]);
```

Comment:
I feel like this is not clean, for the sake of working I have put it here. Have to think on this to make it clean. I also feel like controller assignment can be pushed to base class itself. With the help of this kinda generics. But I have to try